detection_rules/rule_loader.py

```diff
@@ -79,6 +79,7 @@ def load_rules(file_lookup=None, verbose=True, error=True):
     rules = []  # type: list[Rule]
     errors = []
     queries = []
+    query_check_index = []
```

Comment:
```suggestion
    query_check_index = set()
```

Reply:
needs ordering and no need to dedup, so I'll pass on this